packages/wrangler/src/__tests__/dev.test.tsx

```diff
@@ -47,67 +39,123 @@ describe("Dev component", () => {
       `);
     });
   });
+
+  describe("host", () => {
+    it("should resolve a host to its zone", async () => {
+      writeWranglerToml({
+        main: "index.js",
+      });
+      fs.writeFileSync("index.js", `export default {};`);
+      mockGetZones("some-host.com", [{ id: "some-zone-id" }]);
+      await runWrangler("dev --host some-host.com");
+      expect((Dev as jest.Mock).mock.calls[0][0].zone).toEqual({
+        host: "some-host.com",
+        id: "some-zone-id",
+      });
+    });
+
+    it("should read wrangler.toml's dev.host", async () => {
+      writeWranglerToml({
+        main: "index.js",
+        dev: {
+          host: "some-host.com",
+        },
+      });
+      fs.writeFileSync("index.js", `export default {};`);
+      mockGetZones("some-host.com", [{ id: "some-zone-id" }]);
+      await runWrangler("dev");
+      expect((Dev as jest.Mock).mock.calls[0][0].zone.host).toEqual(
+        "some-host.com"
+      );
+    });
+
+    it("should read --route", async () => {
```

Comment:
If `--route` overrides wrangler.toml `routes` it might be worth adding some in here to prove it.